eventuate-tram-spring-reactive-integration-tests/src/test/java/io/eventuate/tram/reactive/integrationtests/ReactiveTramMessagingDuplicateDetectionTest.java

```diff
@@ -0,0 +1,87 @@
+package io.eventuate.tram.reactive.integrationtests;
+
+import io.eventuate.tram.consumer.common.reactive.ReactiveMessageConsumer;
+import io.eventuate.tram.consumer.common.reactive.ReactiveMessageConsumerImplementation;
+import io.eventuate.tram.consumer.common.reactive.ReactiveMessageHandler;
+import io.eventuate.tram.messaging.consumer.MessageSubscription;
+import io.eventuate.tram.messaging.producer.MessageBuilder;
+import io.eventuate.tram.spring.messaging.producer.jdbc.reactive.ReactiveTramMessageProducerJdbcConfiguration;
+import io.eventuate.tram.spring.reactive.consumer.common.ReactiveTramConsumerCommonConfiguration;
+import io.eventuate.tram.spring.reactive.consumer.kafka.EventuateTramReactiveKafkaMessageConsumerConfiguration;
+import io.eventuate.util.test.async.Eventually;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Import;
+import org.springframework.context.annotation.Primary;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.transaction.reactive.TransactionalOperator;
+import reactor.core.publisher.Mono;
+
+import java.util.Collections;
+import java.util.Set;
+import java.util.UUID;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = ReactiveTramMessagingDuplicateDetectionTest.Config.class, properties = "spring.main.allow-bean-definition-overriding=true")
```

Comment:
I think this test can be simplified and improved 

```
For example, imagine a test that uses mocks for EventuateSpringReactiveJdbcStatementExecutor and ReactiveTransactionManager (passed to TransactionalOperator.create())  and verifies that the right thing happens.
```

Specifically, 

* Don't need @SpringBootTest
* Mock EventuateSpringReactiveJdbcStatementExecutor - don't need the database
* Don't mock TransactionalOperator - that implements important behavior
* Mock ReactiveTransactionManager - that's passed to TransactionalOperator.create()

Reply:
Also, simply `new ReactiveTramMessagingDuplicateDetection(...)` and invoke methods on it.
It's so much simpler to test in isolation.

Reply:
What needs to be verified is that the following happens in this order:

1. ReactiveTransactionManager.getReactiveTransaction() 
2. EventuateSpringReactiveJdbcStatementExecutor invoked to execute INSERT
3. message handler is invoked
4. ReactiveTransactionManager.commit()